src/device/aarch64/asm.go

```diff
@@ -0,0 +1,42 @@
+// +build aarch64
+
+package aarch64
+
+// Try to keep go vet from complaining about empty function bodies
+import _ "unsafe"
+
+// Asm runs the given assembly code. The code will be marked as having side effects,
+// as it doesn't produce output and thus would normally be eliminated by the
+// optimizer.
+func Asm(asm string)
+
+// AsmFull runs the given inline assembly. The code will be marked as having side
+// effects, as it would otherwise be optimized away. The inline assembly string
+// recognizes template values in the form {name}, like so:
+//
+//     avr.AsmFull(
+//         "str {value}, {result}",
+//         map[string]interface{}{
+//             "value":  1
+//             "result": &dest,
+//         })
+func AsmFull(asm string, regs map[string]interface{})
+
+// ReadRegister returns the contents of the specified register. The register
+// must be a processor register, reachable with the "mov" instruction.
+func ReadRegister(name string) uintptr
+
+// Run the following system call (SVCall) with 0 arguments.
+func SVCall0(num uintptr) uintptr
```

Comment:
These calls are compiler builtins but have not been implemented in the compiler. Either implement them, or remove this change from the PR to avoid confusion.